docs/gc.md

```diff
@@ -141,15 +141,17 @@ Object ages are tracked for each region with a maximum of 24 possible generation
 
-Over time, transient objects drop out of the live set and viable survivor regions become scarce. This problem is solved by introducing a global mark phase (GMP) to mark and sweep the entire heap. GMP is a hybrid *incremental-concurrent* GC cycle. Each GMP increment is interleaved with a PGC cycle so that regions that are left sparse after sweeping the heap can be collected, cleared, and recycled during and after the GMP.
+Partial GC cycles reclaim free regions by moving objects out of *eden* and older regions. However, moving objects out of *eden* regions can never reclaim all of the consumed memory because some objects from *eden* regions will survive. The partial GC cycle compensates by moving objects to regions that contain older generations so that newer regions can be defragmented. However, over time the overall amount of defragmented memory decreases and records about object liveness in older regions becomes less accurate. Eventually, the work that is done by partial GC cycles to reclaim memory cannot keep pace with memory allocation. Free regions become so scarce that global *mark* operations (sometimes known as the *Global mark phase* (GMP)) are required to build a new record of object liveness across the heap. This record allows a global *sweep* operation to reclaim memory. The global *sweep* operation that is logically associated with the global *mark* operations, runs in the first partial GC cycle after the *mark* operations complete. Because the GC cycle responsible for the global *mark* operations runs concurrently, it might overlap and interleave with a few partial GC cycles. 
 
-With the `balanced` policy, a global GC cycle is rare, occurring only in very tight memory conditions when PGC and GMP operations cannot free enough memory on the heap. These global GCs are explicit GCs, like those invoked by an application calling `System.gc()`.
+With the `balanced` policy, a global GC is sometimes required in addition to the global *mark* operations and PGC. This global GC cycle is rare, occurring only in very tight memory conditions when other GC cycles cannot free enough memory on the heap. This third global GC cycle is an explicit GC cycle, like those invoked by an application calling `System.gc()`.
```

Comment:
This *third* global GC cycle ? Did I miss something here?

Reply:
Let me remove the word "global" - I was referring to the GMP and PGC as the first 2 :-)

Reply:
'is an explicit type' also needs correction. This is more correct:

'This third cycle type is also used with explicit GCs, like those....'

This cycle type when triggered in tight memory condition is considered implicit (since not invoked by user, but by internal GC mechanism).

Reply:
@amicic I don't understand what you are saying. Is is explicit or implicit? One sentence seems to contradict the other.

Reply:
If a GC cycle was triggered by internal mechanisms (allocation failure, taxation threshold reached) then we say it's implicit.
If it's triggered by user (directly by System.gc() or indirectly by for example requesting heap dump, when heap dump will be preceeded by a Global GC) we say it's explicit.

Global GC cycle type can be triggered by other of the two. It can be both explicit or implicit depending how it's triggered.

So, it's not correct to say: 'This third global GC cycle is an explicit GC cycle'. When we say 'is' it implies 1:1 relationship, but to me it's ok to say  'This third cycle type is also used with explicit GCs'. I'm trying to imply 1:n relationship, since this third type is used in implicit scenario, as already stated by 'occurring only in very tight memory conditions '.

Reply:
In that case perhaps that last sentence is redundant, where 'occurring only in very tight memory conditions' implies exactly what causes it and the user does not need the extra information.

Reply:
I think it's ok to have that sentence. We state that if explicit GC is invoked it's Global GC cycle that it will be performed. It brings new information, it's just that the current form is not correct. Again:

'global GC cycle is an explicit GC' is not quite correct, becuase global GC cycle can be also an implicit
'explicit GC is global GC cycle' is correct, since it cannot be PGC cycle or GMP cycle.

Reply:
I think we should discuss explicit/implicit GC in the overview in more depth rather than introducing anything new here. Although it is just extra information, it does not make sense without more words, which are going to detract from the message we are providing in this paragraph. i.e. that this is another global GC that occurs rarely under tight memory restrictions. I've made a note to cover this in the overview topic, which I will be able to construct when this PR finally merges.